src/profile/mfa/mfa_factor_sources/arculus_card_factor_source/arculus_card_factor_source.rs

```diff
@@ -0,0 +1,199 @@
+use crate::prelude::*;
+use crate::prelude::*;
+
+/// An Arculus card, a hierarchal deterministic wallet capable of CAP26 derivation
+/// which users interact with by placing it near their host device, which
+/// communicates with the card over NFC.
+#[derive(
+    Serialize,
+    Deserialize,
+    Clone,
+    PartialEq,
+    Eq,
+    Hash,
+    Debug,
+    derive_more::Display,
+    uniffi::Record,
+)]
+#[serde(rename_all = "camelCase")]
+#[display("{hint} : {id}")]
+pub struct ArculusCardFactorSource {
+    /// Unique and stable identifier of this factor source, stemming from the
+    /// hash of a special child key of the HD root of the mnemonic,
+    /// that is secured by the Arculus Card.
+    pub id: FactorSourceIDFromHash,
+
+    /// Common properties shared between FactorSources of different kinds,
+    /// describing its state, when added, and supported cryptographic parameters.
+    pub common: FactorSourceCommon,
+
+    /// Properties describing a ArculusCardFactorSource to help user disambiguate
+    /// between it and another one.
+    pub hint: ArculusCardHint,
+}
+
+impl From<ArculusCardFactorSource> for FactorSource {
+    fn from(value: ArculusCardFactorSource) -> Self {
+        FactorSource::ArculusCard { value }
+    }
+}
+
+fn new_arculus_with_mwp(
+    mwp: MnemonicWithPassphrase,
+    hint: ArculusCardHint,
+) -> ArculusCardFactorSource {
+    let id = FactorSourceIDFromHash::new_for_arculus(&mwp);
+    let mut source = ArculusCardFactorSource::new(id, hint);
+    source.common.last_used_on = Timestamp::sample();
+    source.common.added_on = Timestamp::sample();
```

Comment:
no this is `sample` and sample values **MUST** have stable values, so we can use expected JSON values for those samples.

Reply:
I see that we are only using these methods from `sample()` & `sample_other()`. However, nothing is stopping a developer from calling this `new_arculus_with_mwp()` (instead of calling `new()`). Maybe we should rename it to `new_sample_arculus()`?

Reply:
Only the extact method of a trait can live inside an `impl MyTrait for MyStruct`.